Code/GraphMol/RGroupDecomposition/RGroupFingerprintScore.cpp

```diff
@@ -238,14 +238,13 @@ static std::mutex groupMutex;
 
 // add an rgroup structure to a bit counts array
 void VarianceDataForLabel::addRgroupData(RGroupData *rgroupData) {
-  if (rgroupData->fingerprint == nullptr) {
 #ifdef RDK_THREADSAFE_SSS
-    const std::lock_guard<std::mutex> lock(groupMutex);
+  const std::lock_guard<std::mutex> lock(groupMutex);
 #endif
-    if (rgroupData->fingerprint == nullptr) {
-      addFingerprintToRGroupData(rgroupData);
-    }
+  if (rgroupData->fingerprint == nullptr) {
+    addFingerprintToRGroupData(rgroupData);
   }
+
```

Comment:
I think this leaves the mutex alive longer than it needs to be.
You can reproduce the original lifetime of the mutex by introducing a local scope:
```
{
#ifdef RDK_THREADSAFE_SSS
  const std::lock_guard<std::mutex> lock(groupMutex);
#endif
  if (rgroupData->fingerprint == nullptr) {
    addFingerprintToRGroupData(rgroupData);
  }
}
```

Reply:
Ok, done